great_expectations/zep/interfaces.py

```diff
@@ -43,3 +62,53 @@ def get_asset(self, asset_name: str) -> DataAsset:
         """Returns the DataAsset referred to by name"""
         # This default implementation will be used if protocol is inherited
         return self.assets[asset_name]
+
+
+class Batch:
+    def __init__(
+        self,
+        datasource: Datasource,
+        data_asset: DataAsset,
+        batch_request: BatchRequest,
+        # BatchDataType is the current type of data in a pre-zep Batch. We might need to define an explicit
+        # interface if we think Datasource implementers will use this as a point of extension.
+        data: BatchDataType,
+    ):
+        """This represents a batch of data.
+
+        This is usually not the data itself but a hook to the data on an external datastore such as
+        a spark or a sql database. An exception exists for pandas or any in-memory datastore.
+        """
+        # These properties are intended to be READ-ONLY
+        self._datasource: Datasource = datasource
+        self._data_asset: DataAsset = data_asset
+        self._batch_request: BatchRequest = batch_request
+        self._data: BatchDataType = data
```

Comment:
Can we annotate these on the class?
Also should we use double underscore name mangling for these given they are meant to be read-only.

```python
class Batch:
        self.__datasource: Datasource
        self.__data_asset: DataAsset
        self.__batch_request: BatchRequest
        self.__data: BatchDataType
        
        def __init__(self, ...):
```